linera-execution/src/lib.rs

```diff
@@ -1088,15 +1088,23 @@ impl Operation {
         Operation::System(operation)
     }
 
+    /// Creates a new user application operation following the `application_id`'s [`Abi`].
     pub fn user<A: Abi>(
         application_id: UserApplicationId<A>,
         operation: &A::Operation,
     ) -> Result<Self, bcs::Error> {
-        let application_id = application_id.forget_abi();
-        let bytes = bcs::to_bytes(&operation)?;
+        Self::user_without_abi(application_id.forget_abi(), operation)
```

Comment:
I find the `forget_abi` a bit weird 🤔 is it to coerce into a different type?

Reply:
Yes, that exactly it. I agree the name could be tweaked in the future, maybe `without_abi` would be more consistent :thinking: